### PR TITLE
OCPQE-19530: Onboard BM IPI jobs in the new infra and add workflows for external LB scenarios

### PR DESCRIPTION
**Pull request by aleskandro**
This PR:
1. enables IPI on ARM64 Baremetal testing @ rdu2 in the new infrastructure. 
2. adds steps and workflows for Baremetal IPI w/ external load balancers
3. Decouple the creation of the HAProxy configuration (templating) from the execution of the load balancer
4. adds other minor fixes and features for the BM IPI steps
5. adds two periodics for BM IPI w/ and w/o external lb.

As BM IPI needs a VM as bootstrap and that VM has to have the same architecture as the cluster being installed, a follow-up PR will be set to use the other auxiliary host as target for creating the VM, hence enabling baremetal ipi on x86 too.

Successful install:
- IPI on BM w/ external LB: https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/46960/rehearse-46960-periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4/1765312847656521728 - [Latest Run (failed with CO network degraded, unrelated??](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/46960/rehearse-46960-periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f7/1765771834613043200)
- IPI on BM w/ internal LB: https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/46960/rehearse-46960-periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-f7/1765717390290587648

Note: the random failures are unrelated to the changes in this PR and are being addressed elsewhere. 

cc @zhaozhanqi

### CODE REVIEW

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

File: ci-operator/step-registry/baremetal/lab/pre/load-balancer/baremetal-lab-pre-load-balancer-commands.sh
Comment (aleskandro) [inline ci-operator/step-registry/baremetal/lab/pre/load-balancer/baremetal-lab-pre-load-balancer-commands.sh:117]: @zhaozhanqi setting the external load balancer IP let the installer set the Ironic IP to that external IP too, meaning that the external LB should expose the ironic APIs too (towards the bootstrap IP first and the master nodes later probably).

See this excerpt from the computed terraform vars:
```
sh-5.1$ cat terraform.platform.auto.tfvars.json  
...
  "ironic_uri": "http://10.10.137.98:6385/v1",
  "inspector_uri": "http://10.10.137.98:5050/v1",
  "bootstrap_os_image": "/alabama/.cache/openshift-installer/image_cache/rhcos-415.92.202311241643-0-qemu.aarch64.qcow2",
  "bridges": [
    {
      "mac": "52:54:00:00:00:49",
      "name": "baremetal"
    },
    {
      "mac": "52:54:00:d2:f3:6a",
      "name": "br--op-sfh1mjrk"
    }
  ],
  "ironic_username": "bootstrap-user",
.....
...
```
- The ironic uri is set to the API VIP, which is the external one.

- Is exposing the ironic APIs through the external lb acceptable in this network scenario?
- Does the internal lb do the same? I.e., Can users reach BM_NETWORK_API_VIP:6385 in non-external lb scenarios? @derekhiggins could give some hint maybe. 

In case this is ok, we will need to document this requirement (the way i did it was just the fastest to test/debug it, I'll consolidate later).

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-f4

**Comment by aleskandro** [comment]
Successful install for baremetal ipi with external lb. https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/46960/rehearse-46960-periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4/1765312847656521728

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-f4

File: ci-operator/step-registry/baremetal/lab/ipi/conf/external-lb/baremetal-lab-ipi-conf-external-lb-commands.sh
Comment (aleskandro) [reply ci-operator/step-registry/baremetal/lab/ipi/conf/external-lb/baremetal-lab-ipi-conf-external-lb-commands.sh:18]: Hi @zhaozhanqi I think we're almost done here, and I settled up the periodic to execute the e2e.

Once that pr merges, you can remove this.... But here, I really would like to move forward as there are many fixes unrelated to your specific work that are worth merging.

I'm running some other testing, although I think we can still fail due to other, hw-related, issues (@sgoveas is working on that other side).

File: ci-operator/step-registry/baremetal/lab/ipi/conf/external-lb/load-balancer/baremetal-lab-ipi-conf-external-lb-load-balancer-commands.sh
Comment (aleskandro) [inline ci-operator/step-registry/baremetal/lab/ipi/conf/external-lb/load-balancer/baremetal-lab-ipi-conf-external-lb-load-balancer-commands.sh:109]: @zhaozhanqi, please be aware that we should document somewhere the need to expose Ironic and the inspector API when installing with the external loadbalancer setting. Unless there is another way to allow these APIs being reachable by the nodes and the installer.

I would only have a further review/approval of this specific change from other devs (either from your team or from the metal one).

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f4

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-f7

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-f7

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f7

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-multi-nightly-baremetal-ipi-ovn-ipv4-external-lb-f7

File: ci-operator/step-registry/baremetal/lab/ipi/conf/external-lb/load-balancer/baremetal-lab-ipi-conf-external-lb-load-balancer-commands.sh
Comment (aleskandro) [reply ci-operator/step-registry/baremetal/lab/ipi/conf/external-lb/load-balancer/baremetal-lab-ipi-conf-external-lb-load-balancer-commands.sh:109]: I'll try to make this land and we can change it later if needed. There are some unrelated fixes here that we need to merge and it's very hard to decouple the Externl LB work now.

**Review by deepsm007 — APPROVED** [review]

**Comment by aleskandro** [comment]
Thanks @deepsm007 . @liangxia I asked Deep to approve for avoiding further conflicts in these changes and because we need them for further debugging about baremetal. If anything will need to change in the test configs, we'll address it later.

**Comment by liangxia** [comment]
> Thanks @deepsm007 . @liangxia I asked Deep to approve for avoiding further conflicts in these changes and because we need them for further debugging about baremetal. If anything will need to change in the test configs, we'll address it later.

I'm fine with that. BTW, no extra changes required from my side.

**Pull request merged**
